  - name: Pull Task Attempt image
    docker_image: name={{task_attempt_docker_image}} api_version=auto source=pull
    register: result
    until: not result.failed is defined or not result.failed
    retries: 10
    delay: 10

  - name: Execute a TaskAttempt
    docker_container:
      name: "{{task_attempt_monitor_container_name}}-{{task_attempt_id}}"
      image: "{{loom_docker_image}}"
      cleanup: yes
      volumes:  "{{[ '/var/run/docker.sock:/var/run/docker.sock', \
                 storage_root~':'~internal_storage_root ]}}"
      api_version: auto
      network_mode: host
      command: /bin/bash -c 'loom-task-monitor --task_attempt_id {{task_attempt_id}} --server_url {{server_url}} --log_level {{log_level}}{% if token %} --token {{token}}{% endif %}'
      log_driver: fluentd
      log_options:
        fluentd-address: "{{fluentd_address}}"
        fluentd-async-connect: "{{fluentd_async_connect}}"
        tag: "{{fluentd_tag}}"
